semgrep/semgrep/test.py

```diff
@@ -34,14 +35,28 @@
 SAVE_TEST_OUTPUT_JSON = "semgrep_runs_output.json"
 SAVE_TEST_OUTPUT_TAR = "semgrep_runs_output.tar.gz"
 
+COMMENT_SYNTAXES = (("#", "\n"), ("//", "\n"), ("<!--", "-->"), ("(*", "*)"))
+SPACE_OR_NO_SPACE = ("", " ")
+TODORULEID = "todoruleid"
+RULEID = "ruleid"
+TODOOK = "todook"
+OK = "ok"
 
-def normalize_rule_id(line: str) -> str:
+
+def normalize_rule_ids(line: str) -> Set[str]:
     """
     given a line like `     # ruleid:foobar`
     or `      // ruleid:foobar`
     return `foobar`
     """
-    return line.strip().split(":")[1].strip().split(" ")[0].strip()
+    _, rules_text = line.strip().split(":")
+    rules_text = rules_text.strip()
+    rules = rules_text.split(",")
+    # strictly remove ending HTML comment
```

Comment:
If you want to generalize this so we don't need to remember to update these as well we could do something like:

```python
rules = [
    rule.strip() if end == "\n" else rule.strip().replace(end, "")
    for start, end in COMMENT_SYNTAXES
]
```

Then we could also remove the `rule.strip()` call in the `return` statement.